### PR TITLE
Add disabled property on menu items

### PR DESCRIPTION
**Pull request by oorestisime**
Fix: #2766
#### What does this PR do?
Add disable property on menu items

#### What testing has been done on this PR?
storybook
#### What are the relevant issues?
#2766 
#### Do the grommet docs need to be updated?
right now the items are not documented. wonder whether we should
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
yes

### CODE REVIEW

File: src/js/components/Menu/Menu.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ class Menu extends Component {
                       }}
                       active={activeItemIndex === index}
                       hoverIndicator="background"
-                      disabled={!item.onClick && !item.href}
+                      disabled={item.disabled || (!item.onClick && !item.href)}
```
Comment (ericsoderberghp) [inline src/js/components/Menu/Menu.js]: I'm thinking we should let Button handle the `onClick` and `href` aspects directly and just pass disabled directly through. Would that work?

Comment (oorestisime) [reply src/js/components/Menu/Menu.js]: you mean if there is no onClick or href on button to be disabled?

Comment (ericsoderberghp) [reply src/js/components/Menu/Menu.js]: If the caller adds `disabled` to the item, then we should pass that to Button. If the caller doesn't pass `disabled` in the item, I don't think Menu should construct another value to pass to Button.

Comment (oorestisime) [reply src/js/components/Menu/Menu.js]: ok done! i wonder whether this should be noted as a breaking change. people might have relied on that behavior

**Pull request merged**